pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -293,10 +293,20 @@ func (at *ActivityTree) insertEvent(event *model.Event, dryRun bool, generationT
 		return false, fmt.Errorf("invalid event: %s", err)
 	}
 
-	node, newProcessNode, err := at.CreateProcessNode(event.ProcessCacheEntry, generationType, dryRun, resolvers)
+	var createProcessNodeDryRun bool
+	if dryRun || !insertMissingProcesses {
+		createProcessNodeDryRun = true
+	} else {
+		createProcessNodeDryRun = false
+	}
+	node, newProcessNode, err := at.CreateProcessNode(event.ProcessCacheEntry, generationType, createProcessNodeDryRun, resolvers)
 	if err != nil {
 		return false, err
 	}
+	if newProcessNode && !dryRun && !insertMissingProcesses {
```

Comment:
Same here ? 🤔

Reply:
I just added some comments. Hopefully it's more clear now!